mpqp/core/instruction/measurement/basis.py

```diff
@@ -33,27 +35,48 @@ class Basis:
             ``basis_vectors``'s dimensions.
 
     Example:
-        >>> Basis([np.array([1,0]), np.array([0,-1])]).pretty_print()
+        >>> custom_basis = Basis([np.array([1,0]), np.array([0,-1])], symbols=("↑", "↓"))
+        >>> custom_basis.pretty_print()
         Basis: [
             [1, 0],
             [0, -1]
         ]
+        >>> circ = QCircuit([X(0), H(1), CNOT(1, 2), Y(2)])
+        >>> circ.add(BasisMeasure([0, 1, 2], basis=custom_basis, shots=10000))
+        >>> print(run(circ, IBMDevice.AER_SIMULATOR)) # doctest: +SKIP
+        Result: None, IBMDevice, AER_SIMULATOR
+         Counts: [0, 0, 0, 0, 0, 4936, 5064, 0]
+         Probabilities: [0, 0, 0, 0, 0, 0.4936, 0.5064, 0]
+         Samples:
+          State: ↓↑↓, Index: 5, Count: 4936, Probability: 0.4936
+          State: ↓↓↑, Index: 6, Count: 5064, Probability: 0.5064
+         Error: None
 
     """
 
     def __init__(
         self,
         basis_vectors: list[npt.NDArray[np.complex64]],
         nb_qubits: Optional[int] = None,
+        symbols: Optional[tuple[str, str]] = None,
+        basis_vectors_labels: Optional[list[str]] = None,
     ):
-        # TODO : add the possibility to give the symbols for the '0' and '1' of
-        # the custom basis. This should then appear in the Sample
-        # binary_representation of the basis state. For instance in the Hadamard
-        # basis, the symbols will be '+' and '-'. If the user wants '↑' and '↓'
-        # for his custom basis, when we print samples we would have something
-        # like:
-        # State: ↑↑↓, Index: 1, Count: 512, Probability: 0.512
+        if symbols is not None and basis_vectors_labels is not None:
+            raise ValueError(
+                "You can only specify either symbols or basis_vectors_labels, not both."
+            )
+
+        if symbols is None:
```

Comment:
good catch, you could use GitHub's `suggestion` feature for tiny changes like this one

Reply:
but in fact we always first check that `basis_vectors_labels` is not `None`, so this is not that relevant. We could add this check but it is rather insignificant (convince me otherwise :p )

Reply:
If we always check that before, then we good.